src/IO/ReadSettings.h

```diff
@@ -101,6 +102,12 @@ struct ReadSettings
     /// Don't populate cache when the read is not part of query execution (e.g. background thread).
     bool avoid_readthrough_cache_outside_query_context = true;
 
+    bool enable_page_cache = false;
```

Comment:
It's a bit confusing to have here both `enable_page_cache` and `force_enable_page_cache` and `enable_page_cache` not being updated by `Context::getReadSettings` like all other settings, I saw why it is needed - because `enable_page_cache` is set with a special logic in `DiskObjectStorage::readFile`, but again why not just have a single setting only as user setting without this complication?